src/auth/verification.service.ts

```diff
@@ -0,0 +1,70 @@
+import * as crypto from 'crypto';
+import base64url from 'base64url';
+import { Injectable, GoneException, BadRequestException, NotFoundException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { MailerService } from 'src/mailer/mailer.service';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { EmailVerificationDto } from './dto';
+
+@Injectable()
+export class VerificationSerivce {
+  constructor(
+    private readonly mailerService: MailerService,
+    private readonly config: ConfigService,
+    private readonly prismaService: PrismaService
+  ) {}
+
+  send(email: string): string {
+    const hash = crypto.randomBytes(32).toString('hex');
+    const TWO_DAYS = 1000 * 60 * 60 * 24 * 2;
```

Comment:
Utilize constants for magic numbers (like TWO_DAYS) and extract configuration values from the service instead of hardcoding them directly in the methods
`private readonly TWO_DAYS = 1000 * 60 * 60 * 24 * 2;`